go/pkg/command/command.go

```diff
@@ -45,11 +45,27 @@ type InputExclusion struct {
 	Type InputType
 }
 
+// VirtualInput represents an input that does not actually exist as a file on disk, but we want
+// to stage it as a file on disk for the command execution.
+type VirtualInput struct {
+	// The path for the input file to be staged at, relative to the ExecRoot.
+	Path string
+
+	// The byte contents of the file to be staged.
+	Contents []byte
+
+	// Whether the file should be staged as executable.
+  IsExecutable bool
```

Comment:
indentation seems broken here

Reply:
Indeed! Forgot to run lint and go fmt. I should really add it to presubmit...
Done.